### ISSUE TITLE
snowflake_grant_privileges_to_role doesn't work with for_each role

### ISSUE THREAD

**Issue by chrisluedtke** [Snowflake-Labs/terraform-provider-snowflake#2423]
### Terraform CLI and Provider Versions

Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/aidanmelen/snowsql v0.3.0
+ provider registry.terraform.io/hashicorp/azurerm v3.50.0
+ provider registry.terraform.io/hashicorp/local v2.4.1
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/tls v4.0.5
+ provider registry.terraform.io/snowflake-labs/snowflake v0.75.0

### Terraform Configuration

```terraform
resource "snowflake_grant_privileges_to_role" "usage_source_db" {
  for_each   = toset([snowflake_role.developer.name, snowflake_role.loader.name])
  privileges = ["USAGE"]
  role_name  = each.key
  on_account_object {
    object_type = "DATABASE"
    object_name = snowflake_database.source.name
  }
}

resource "snowflake_grant_privileges_to_role" "usage_all_source_schemas" {
  for_each = toset([
    snowflake_role.developer.name,
    snowflake_role.loader.name,
  ])
  privileges = ["USAGE"]
  role_name  = each.key
  on_schema {
    all_schemas_in_database = snowflake_database.source.name
  }
}

resource "snowflake_grant_privileges_to_role" "usage_future_source_schemas" {
  for_each = toset([
    snowflake_role.developer.name,
    snowflake_role.loader.name,
  ])
  privileges = ["USAGE"]
  role_name  = each.key
  on_schema {
    future_schemas_in_database = snowflake_database.source.name
  }
}

resource "snowflake_grant_privileges_to_role" "loader_all_source_schemas" {
  privileges = ["CREATE TABLE", "MODIFY", "CREATE PIPE", "CREATE STAGE"]
  role_name  = snowflake_role.loader.name
  on_schema {
    all_schemas_in_database = snowflake_database.source.name
  }
}

resource "snowflake_grant_privileges_to_role" "loader_future_source_schemas" {
  privileges = ["CREATE TABLE", "MODIFY", "CREATE PIPE", "CREATE STAGE"]
  role_name  = snowflake_role.loader.name
  on_schema {
    future_schemas_in_database = snowflake_database.source.name
  }
}
```

### Expected Behavior

Both developer and loader roles have access to all specified schemas.

### Actual Behavior

Only the "loader" role has access to the schemas.

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

High

**Comment by chrisluedtke** [comment]
This issue appears to be resolved after upgrading the provider to 0.84.1

**Comment by chrisluedtke** [comment]
This issue appears to be resolved after upgrading the provider to 0.84.1
